include/dgl/runtime/ndarray.h

```diff
@@ -229,6 +231,9 @@ class NDArray {
    */
   DGL_DLL static NDArray Empty(
       std::vector<int64_t> shape, DGLDataType dtype, DGLContext ctx);
+
+  DGL_DLL static NDArray PinnedEmpty(
```

Comment:
Ditto.

Reply:
Yeah.. it makes sense and it's more like pytorch's passing tensor options to `empty` function. 
Here, since we may only have one option `pin_memory` and other empty-like function (`empty_shared`) is also separated from `empty`, maybe it's better to put it in another PR, in case we have more `empty` like functions in future.